var/spack/repos/builtin/packages/hcol/package.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Hcol(Package):
+    """This is the SPIRAL package for the Hybrid Control Operator Language
+    (HCOL)."""
+
+    homepage = "https://https://spiral.net"
+    url      = "https://github.com/spiral-software/spiral-package-hcol/archive/1.0.0.zip"
+
+    maintainers = ['spiralgen']
+    extends('spiral')
+
+    version('1.0.0', sha256='9a95574e2b061d03d264c32dbf733e893017d1644b6486c7a8a55a3b24783f58')
+
+    # HCOL package is an extension for Spiral.  Install the files in their own
+    # prefix.  This folder needs to be linked to spiral so that spiral can
+    # import/use the package
```

Comment:
@scheibelp @alalazo @becker33 does one of you know the proper way to implement `activate` and `deactivate` in an extendable package?

Reply:
There is a default implementation that may work "out-of-the-box" for global activations and views. Based on the description here, it sounds like `hcol/spiral` would work fine with the default implementation. `r` for example doesn't have to override the default; `python` does to maintain a Python-specific file that is used to search for dependencies.

If you encounter any issues with global/view activation, you can post the error message here (or in a separate issue) and I can suggest how the `spiral` package would need to be updated.

Reply:
A little background may help... We're adding spiral to spack to make it more easily available for broader use in the ECP project.  Spiral is a 20-year-old package with certain baked in conventions and an existing structure that would be disruptive to upend.  Spiral packages (in this instance hcol; and hcol is the first of several) must be installed in the namespaces/packages folder so that they can be found, loaded/imported to spiral.  Installing hcol in its own prefix is acceptable so long as we can create a link from spiral/namespaces/packages -> hcol.  

spack activate wants to make links to every file in the hcol package (I see this as overkill) but it places them in the root of spiral (rather than namespaces/packages).   Is there a way to override the default spack activate/deactivate methods to simply create the link needed?  If not we know (I demonstrated) that creating the symlink during the install phase for hcol works and spiral sees and can use the extension package.

Reply:
It sounds like you would want to create a `SpiralPackage` in `lib/spack/spack/build_systems` that overrides `view_destination`:

```
def view_destination(selef, view):
    return os.path.join(view.get_projection_for_spec(self.spec), "namespaces", "packages")
```

the `aspell` package has a similar pattern (see: https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/aspell_dict.py - although for that package, the destination is determined by invoking the executable).

You can restrict the files that are added by overriding `add_files_to_view`, but IMO as long as it doesn't break anything, it's easier to just let Spack link everything in `hcol`.

Reply:
Will install hcol in <prefix>/namespaces/packages/hcol.  We can use the standard spack activate hcol to create the necessary linkage between spiral and hcol.  hcol package definition updated accordingly.

Reply:
Put hcol tree in ,prefix>/namespaces/packages/hcol; now vanilla 'spack activate' can link the package so spiral can use.  No other changes required.